apps/condo/domains/billing/schema/resolvers/utils.js

```diff
@@ -53,10 +58,67 @@ const normalizePropertyGlobalId = (rawFiasCode) => {
 
 const sortPeriodFunction = (periodA, periodB) => (dayjs(periodA, 'YYYY-MM-DD').isAfter(dayjs(periodB, 'YYYY-MM-DD')) ? 1 : -1)
 
+const buildLastReportForBillingContext = async (receipts, { dv, sender, billingContext }) => {
+    if (!receipts.length) return
+
+    const { keystone: context } = getSchemaCtx('BillingIntegrationOrganizationContext')
+    const receiptsCategories = [...new Set(receipts.map(({ category }) => category).filter(Boolean))]
+    const receiptsPeriods = [...new Set(receipts.map(({ period }) => period).filter(Boolean))]
+
+    const newestPeriodFromReceipts = receiptsPeriods.sort(sortPeriodFunction).pop()
+    const newerReceiptsCount = await BillingReceipt.count(context, {
```

Comment:
You can speed this up a lot:
1. `count` vs `getAll` with `first`. What is period is 6 months ago? When you need to count ALL receipts to compute count. Do you actually use count? What you actually need it, is "Is there at least 1 new receipt?". So use first, so you database can return as soon as found 1 receipt to avoid full scan
2. `getAll` vs `find` vs `itemsQuery`. getAll with resolve all relations, so it's not performant, `find` returns all rows, so it does not support query. Items query do, so use it instead